oneandone/src/main/java/org/apache/jclouds/oneandone/rest/compute/config/OneAndOneComputeServiceContextModule.java

```diff
@@ -0,0 +1,179 @@
+import org.apache.jclouds.oneandone.rest.compute.OneandoneComputeServiceAdapter;
+import org.apache.jclouds.oneandone.rest.compute.function.HardwareFlavourToHardware;
+import org.apache.jclouds.oneandone.rest.compute.function.HddToVolume;
+import org.apache.jclouds.oneandone.rest.compute.function.ImageToImage;
+import org.apache.jclouds.oneandone.rest.compute.function.ServerToNodeMetadata;
+import static org.apache.jclouds.oneandone.rest.config.OneAndOneProperties.POLL_MAX_PERIOD;
+import static org.apache.jclouds.oneandone.rest.config.OneAndOneProperties.POLL_PERIOD;
+import static org.apache.jclouds.oneandone.rest.config.OneAndOneProperties.POLL_PREDICATE_SERVER;
+import static org.apache.jclouds.oneandone.rest.config.OneAndOneProperties.POLL_PREDICATE_SERVER_RUNNING;
+import static org.apache.jclouds.oneandone.rest.config.OneAndOneProperties.POLL_TIMEOUT;
+import org.apache.jclouds.oneandone.rest.domain.HardwareFlavour;
+import org.apache.jclouds.oneandone.rest.domain.Hdd;
+import org.apache.jclouds.oneandone.rest.domain.Server;
+import org.apache.jclouds.oneandone.rest.domain.Types;
+import org.jclouds.compute.ComputeServiceAdapter;
+import org.jclouds.compute.config.ComputeServiceAdapterContextModule;
+import org.jclouds.compute.domain.Hardware;
+import org.jclouds.compute.domain.Image;
+import org.jclouds.compute.domain.NodeMetadata;
+import org.jclouds.compute.domain.Volume;
+import org.jclouds.compute.domain.internal.ArbitraryCpuRamTemplateBuilderImpl;
+import org.jclouds.compute.domain.internal.TemplateBuilderImpl;
+import org.jclouds.domain.Location;
+import org.jclouds.location.suppliers.ImplicitLocationSupplier;
+import org.jclouds.location.suppliers.implicit.OnlyLocationOrFirstRegionOptionallyMatchingRegionId;
+import static org.jclouds.util.Predicates2.retry;
+
+public class OneAndOneComputeServiceContextModule extends
+        ComputeServiceAdapterContextModule<Server, HardwareFlavour, org.apache.jclouds.oneandone.rest.domain.Image, Location> {
+
+   @SuppressWarnings("unchecked")
+   @Override
+   protected void configure() {
+      super.configure();
+
+      bind(ImplicitLocationSupplier.class).to(OnlyLocationOrFirstRegionOptionallyMatchingRegionId.class).in(Scopes.SINGLETON);
+
+      bind(new TypeLiteral<ComputeServiceAdapter<Server, HardwareFlavour, org.apache.jclouds.oneandone.rest.domain.Image, Location>>() {
+      }).to(OneandoneComputeServiceAdapter.class);
+
+      bind(TemplateBuilderImpl.class).to(ArbitraryCpuRamTemplateBuilderImpl.class);
+
+      bind(new TypeLiteral<Function<Server, NodeMetadata>>() {
+      }).to(ServerToNodeMetadata.class);
+
+      bind(new TypeLiteral<Function<org.apache.jclouds.oneandone.rest.domain.Image, Image>>() {
+      }).to(ImageToImage.class);
+
+      bind(new TypeLiteral<Function<Hdd, Volume>>() {
+      }).to(HddToVolume.class);
+
+      bind(new TypeLiteral<Function<HardwareFlavour, Hardware>>() {
+      }).to(HardwareFlavourToHardware.class);
+   }
```

Comment:
If the public key configured by the user can be injected using the API (as it seems), then add [this binding](https://github.com/jclouds/jclouds/blob/master/providers/google-compute-engine/src/main/java/org/jclouds/googlecomputeengine/compute/config/GoogleComputeEngineServiceContextModule.java#L121), so jclouds does not try to upload it as part of a bootstrap ssh script.